src/libclipper/include/clipper/selection_policies.hpp

```diff
@@ -0,0 +1,287 @@
+#ifndef CLIPPER_LIB_SELECTION_POLICY_H
+#define CLIPPER_LIB_SELECTION_POLICY_H
+
+#include <map>
+#include <memory>
+#include <unordered_map>
+
+#include "datatypes.hpp"
+#include "task_executor.hpp"
+
+namespace clipper {
+//// State Data Structure
+using ModelInfo = std::unordered_map<std::string, double>;
+using Map = std::unordered_map<VersionedModelId, ModelInfo,
+                               std::function<size_t(const VersionedModelId&)>>;
+// Exp3: sum of weights; each model has "weight", "max loss"
```

Comment:
Let's make the state objects classes instead of nested maps and pairs. Then we can give all the fields nice names and avoiding referencing things as `first` and `second`.